lib/message/encrypt.js

```diff
@@ -30,40 +57,34 @@ export default async function encryptMessage({
     }
 
     if (detached) {
-        // Add cleartext signature for backward compatibility
-        const { message } = sanitizedOptions;
-        if (message.fromStream) {
-            // TODO clone input stream -- sanitizedOptions.message.fromStream ? await readMessage({ clone(<input data>) }) : sanitizedOptions.message;
-            throw new Error('Encrypting and detached-signing a message created from a stream is not supported yet');
+        if (!armor && isStream(textData || binaryData)) {
+            throw new Error('Unsupported detached signature when streaming data and requesting non-armored result');
         }
-        const signature = await sign({
-            message,
+
+        const detachedSignatureBinary = await sign({
+            message: await createMessageFromPassiveStreamClone({ [dataType]: data }),
```

Comment:
I think the code becomes simpler if you inline the above functions, e.g.
```suggestion
            message: await createMessage({
                [dataType]: isStream(data) ? passiveClone(data) : data,
                ...
            }),
```
(Also, date and options are not passed here, but they are above, probably you wanted to add them here?)

Reply:
I don't think we need to pass the `date` since encrypt/sign overwrite it? `options` was intended for `createMessage` options in case we kept exposing the function, but now it's not needed.

Reply:
Actually, no, encrypt/sign don't overwrite it. The date passed to encrypt/sign is used for the signature creation date, but the message creation date still comes from `createMessage`. So it's probably better to still pass it